### ISSUE TITLE
Azure AD PowerShell module deprecated March 30, 2024

### ISSUE THREAD

**Issue by johnsonkenneth** [MicrosoftDocs/microsoft-365-docs#12632]
The Azure AD PowerShell module is due to be deprecated on March 30, 2024. Please provide an updated script on this page that uses the MS Graph PowerShell before that deprecation date.

We believe that this is currently possible using the Beta Graph PowerShell commands, but we would really like to see an update to the script that uses GA MS Graph PowerShell commands. These are the commands in the script that have equivalents in the Beta MS Graph PowerShell module at this time.
Connect-AuzreAD -> Connect-MgGraph
Get-AzureADDirectorySetting --> Get-MgBetaDirectorySetting
New-AzureADDirectorySetting --> New-MgBetaDirectorySetting
Set-AzureADDirectorySetting --> Update-MgBetaDirectorySetting
Get-AzureADDirectorySettingTemplate --> Get-MgBetaDirectorySettingTemplate
Get-AzureADGroup --> Get-MgGroup

It would also be nice if you provide an alternative way to make these changes with the MS Graph REST APIs. It looks like this can be done only with the Beta APIs at this time. Of course, we would prefer to use the GA APIs and not the Beta APIs.
https://graph.microsoft.com/beta/settings - Reference doc: [Create settings - Microsoft Graph beta | Microsoft Learn](https://learn.microsoft.com/en-us/graph/api/group-post-settings?view=graph-rest-beta&tabs=http)
https://graph.microsoft.com/beta/directorySettingTemplates - Reference doc: [List directorySettingTemplates - Microsoft Graph beta | Microsoft Learn](https://learn.microsoft.com/en-us/graph/api/directorysettingtemplate-list?view=graph-rest-beta&tabs=http)

Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 445c0a2e-1a22-d90f-0c73-0edf34f2b99a
* Version Independent ID: d534ac69-134d-0198-fcb9-e19d1ed44d4b
* Content: [Manage who can create Microsoft 365 Groups](https://learn.microsoft.com/en-us/microsoft-365/solutions/manage-creation-of-groups?view=o365-worldwide&viewFallbackFrom=o365-worldwide.)
* Content Source: [microsoft-365/solutions/manage-creation-of-groups.md](https://github.com/MicrosoftDocs/microsoft-365-docs/blob/public/microsoft-365/solutions/manage-creation-of-groups.md)
* Service: **o365-solutions**
* GitHub Login: @MikePlumleyMSFT
* Microsoft Alias: **mikeplum**

**Comment by yogkumgit** [comment]
@MikePlumleyMSFT Please help us in resolving this issue. Thanks

**Comment by suthAu** [comment]
Yes! This is very much needed especially as this is an a Microsoft exam topic.@Micrsoft PLEASE include this in your official documentation!!!

**Comment by suthAu** [comment]
I have tried to swap out the PowerShell cmdlets for the Microsoft Graph for PowerShell SDK cmdlets in the script on the [Manage who can create Microsoft 365 Groups](https://learn.microsoft.com/en-us/microsoft-365/solutions/manage-creation-of-groups?view=o365-worldwide&viewFallbackFrom=o365-worldwide.) help article, but I get the error "New-MgBetaDirectorySetting: A parameter cannot be found that matches parameter name 'DirectorySettings'.". Can anyone help me resolve this ... tell me if the code is completely wrong or if there is a replacement for the parameter 'DirectorySettings'??

### Code the manage who can create Microsoft 365 Groups

$GroupName = "Group Creators"
$AllowGroupCreation = $False

Connect-MgGraph -Scopes "Directory.ReadWrite.All"

$settingsObjectID = (Get-MgBetaDirectorySetting | Where-object -Property Displayname -Value "Group.Unified" -EQ).id
if(!$settingsObjectID)
{
    $template = Get-MgBetaDirectorySettingTemplate | Where-object {$_.displayname -eq "group.unified"}
    $settingsCopy = $template.CreateDirectorySetting()
    New-MgBetaDirectorySetting -DirectorySetting $settingsCopy
    $settingsObjectID = (Get-MgBetaDirectorySetting | Where-object -Property Displayname -Value "Group.Unified" -EQ).id
}

$settingsCopy = Get-MgBetaDirectorySetting -DirectorySettingId "$settingsObjectID"
$settingsCopy["EnableGroupCreation"] = $AllowGroupCreation

if($GroupName)
{
  $settingsCopy["GroupCreationAllowedGroupId"] = (Get-MgGroup -Filter "DisplayName eq '$GroupName'").objectid
} else {
$settingsCopy["GroupCreationAllowedGroupId"] = $GroupName
}
New-MgBetaDirectorySetting -Id $settingsObjectID -DirectorySettings $settingsCopy

(Get-MgBetaDirectorySetting -DirectorySettingId "$settingsObjectID").Values

**Comment by officedocspr5** [comment]
To make it easier for you to submit feedback on articles on learn.microsoft.com, we're transitioning our feedback system from GitHub Issues to a new experience. 

As part of the transition, this GitHub Issue will be moved to a private repository. We're moving Issues to another repository so we can continue working on Issues that were open at the time of the transition. When this Issue is moved, you'll no longer be able to access it.

If you want to provide additional information before this Issue is moved, please update this Issue before December 15th, 2023.

With the new experience, you no longer need to sign in to GitHub to enter and submit your feedback. Instead, you can choose directly on each article's page whether the article was helpful. Then you can then choose one or more reasons for your feedback and optionally provide additional context before you select **Submit**.

Here's what the new experience looks like.

**Note**: The new experience is being rolled out across learn.microsoft.com in phases. If you don't see the new experience on an article, please check back later.

First, select whether the article was helpful:

![Image showing a dialog asking if the article was helpful with yes and no answers.](https://learn.microsoft.com/Office/media/internal/new-feedback-yes-no.jpg)

Then, choose at least one reason for your feedback and optionally provide additional details about your feedback:

| Article was helpful | Article was unhelpful |
|:--|:---|
|![Image showing a dialog asking how the article was helpful with several options.](https://learn.microsoft.com/Office/media/internal/new-feedback-helpful.jpg) | ![Image showing a dialog asking how the article wasn't helpful with several options.](https://learn.microsoft.com/Office/media/internal/new-feedback-unhelpful.jpg)|

Finally, select **Submit** and you're done!
